arbor/threading/threading.hpp

```diff
@@ -179,7 +208,18 @@ struct parallel_for {
         for (int i = left; i < right; ++i) {
           g.run([=] {f(i);});
         }
-        g.wait();
+        auto ex = g.wait();
+
+        if(ex) {
+            try
+            {
+                std::rethrow_exception(ex);
+            }
+            catch (const std::exception &e)
+            {
+                throw std::runtime_error(e.what());
```

Comment:
Instead of `throw::runtime_error(e.what())` should it be `std::rethrow_exception(e)`?

Reply:
rethrow_exception takes an exception_ptr, so that won't work I think. Should I throw something other than a runtime_error?

Reply:
Sorry, I didn't read that code very well when reviewing it. I see that you are already trying `std::rethrow_exception`.
Then, I wonder if just `throw;` will rethrow `e`?